File: Behavioral/State/JavaScript/redLight.js
// redLight.js

const TrafficLightState = require("./trafficLightState");

class RedLight extends TrafficLightState {
  handle(trafficLight) {
    console.log("Red Light - STOP!");
    
    // Lazy import to avoid circular dependency
    const GreenLight = require("./greenLight");
    trafficLight.setState(new GreenLight()); // Transition to Green
  }
}

module.exports = RedLight;

File: Behavioral/State/Python/red_light.py
# red_light.py

from traffic_light_state import TrafficLightState
import green_light

class RedLight(TrafficLightState):
  def handle(self, traffic_light):
    print("Red Light - STOP!")
    traffic_light.set_state(green_light.GreenLight())  # Transition to Green
